### ISSUE TITLE
cannot import name 'authenticated_userid' from 'pyramid.security'

### ISSUE THREAD

**Issue by Peter-Slump** [stevearc/pypicloud#274]
Since today I get the message `cannot import name 'authenticated_userid' from 'pyramid.security'` when checking the Pyramids source code you can see that the `authenticated_userid` function is deprecated and removed from Pyramid 1.8

Pypicloud makes use of deprecated functionality in Pyramid but doens't limit the version of Pyramid. So either the usage of this function should be removed or the version of Pyramid should be limited to < 1.8

**Comment by Peter-Slump** [comment]
Ah I see that Pyramid 2.0 is just released a couple of hours ago 😅 that would probably the reason that this error pops up now.

**Comment by stevearc** [comment]
I released version 1.2.0 which pins the version of Pyramid to `<2.0`. I'll try to get the docker container updated tomorrow, but no promises. Cryptography is being a difficult dependency again.

**Comment by aperuru** [comment]
I'm running into similar issue, any progress on the issue?
```
2021-06-07 23:31:48Jun 8 06:31:48 ip-172-31-21-37 syslog-ng[15]: syslog-ng shutting down; version='3.13.2'
2021-06-07 23:31:48*** /etc/my_init.d/pypicloud-uwsgi.sh failed with status 22
2021-06-07 23:31:48*** Killing all processes...
2021-06-07 23:31:48return self.name_resolver.maybe_resolve(dotted)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/path.py", line 327, in maybe_resolve
2021-06-07 23:31:48return self._resolve(dotted, package)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/path.py", line 334, in _resolve
2021-06-07 23:31:48return self._zope_dottedname_style(dotted, package)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/path.py", line 389, in _zope_dottedname_style
2021-06-07 23:31:48__import__(used)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pypicloud/auth.py", line 11, in <module>

2021-06-07 23:31:48from pyramid.security import Everyone, authenticated_userid
2021-06-07 23:31:48ImportError: cannot import name 'authenticated_userid'

2021-06-07 23:31:48c(configurator)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pypicloud/__init__.py", line 80, in includeme
2021-06-07 23:31:48config.include("pypicloud.auth")
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/config/__init__.py", line 632, in include
2021-06-07 23:31:48c = self.maybe_dotted(callable)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/config/__init__.py", line 740, in maybe_dotted
2021-06-07 23:31:48return context.create()
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 715, in create
2021-06-07 23:31:48return self.object_type.invoke(self)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 152, in invoke
2021-06-07 23:31:48return fix_call(context.object, context.global_conf, **context.local_conf)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/paste/deploy/util.py", line 55, in fix_call
2021-06-07 23:31:48val = callable(*args, **kw)
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pypicloud/__init__.py", line 199, in main
2021-06-07 23:31:48config.include("pypicloud")
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/config/__init__.py", line 666, in include
2021-06-07 23:31:48Traceback (most recent call last):
2021-06-07 23:31:48File "/usr/local/lib/python3.6/dist-packages/pyramid/path.py", line 387, in _zope_dottedname_style
2021-06-07 23:31:48found = getattr(found, n)

2021-06-07 23:31:48AttributeError: module 'pypicloud' has no attribute 'auth'
```

**Comment by aperuru** [comment]
Why is changelog not update here: https://pypicloud.readthedocs.io/en/latest/changes.html
Looks like the build failed, can you please fix it @stevearc ?

**Comment by stevearc** [comment]
Documentation build fixed. I also released 1.2.2 which removes the version pinning and upgrades to pyramid 2.0.

**Comment by stevearc** [comment]
This is unrelated to the original issue. Please open another issue with this report.

The error is happening because the package upload handler can't find the `content` parameter of the POST request. Make sure that you're performing the upload correctly (example in the documentation: https://pypicloud.readthedocs.io/en/latest/topics/getting_started.html#uploading-packages). If nothing looks off there, try inspecting the raw request being sent up and make sure that it contains a `content` parameter. You can find documentation about the endpoint as well as a `curl` command to experiment with here: https://pypicloud.readthedocs.io/en/latest/topics/api.html#post-simple
